mon-pix/app/routes/campaigns/send-profile.js

```diff
@@ -7,19 +7,22 @@ export default class CampaignsSendProfileRoute extends Route {
   @service store;
 
   async model(params) {
-    const userId = this.currentUser.user.id;
+    const user = this.currentUser.user;
 
     const campaigns = await this.store.query('campaign', { filter: { code: params.campaign_code } });
     const campaign = campaigns.get('firstObject');
     return RSVP.hash({
       campaign,
-      campaignParticipation: this.store.queryRecord('campaignParticipation', { campaignId: campaign.id, userId }),
+      campaignParticipation: this.store.queryRecord('campaignParticipation', { campaignId: campaign.id, userId: user.id }),
+      user
     });
   }
 
-  async afterModel({ campaign, campaignParticipation }) {
+  async afterModel({ campaign, campaignParticipation, user }) {
     if (!campaignParticipation) {
       return this.replaceWith('campaigns.start-or-resume', campaign.code);
     }
+    user.hasMany('scorecards').reload();
+    user.belongsTo('pixScore').reload();
```

Comment:
on a vu que ne pas avoir de `return` pouvait perturber ember. Du coup :
```suggestion
    return user.belongsTo('pixScore').reload();
```

Reply:
On pourrait faire un Promise.all des deux, non ?

Reply:
Peut-être plus clair d'utiliser `await` :
```js
    await user.hasMany('scorecards').reload();
    await user.belongsTo('pixScore').reload();
```

Reply:
Les deux n'ont pas de liens entre deux. Pourquoi les attendre l'un après l'autre ?

Reply:
C'est vrai qu'on peut lancer les deux en parallèle, pour l'utilisateur individuel c'est le mieux pour réduire la latence. Mais ces temps-ci j'ai tendance à pencher du côté du collectif 😉 et choisir ce qui étale la charge sur le serveur, au prix d'un peu plus de latence pour un utilisateur. (On pourrait aussi parler du fait que les deux font essentiellement le même calcul en remontant les mêmes données de la base, mais c'est un autre sujet…)

En l'occurrence, ça ne fera probablement pas de grosse différence, donc pourquoi pas un `await Promise.all[user.hasMany('scorecards').reload(), user.belongsTo('pixScore').reload()]`.

Une autre considération éventuelle est qu'on ne devrait pas avoir besoin d'attendre la résolution de ces promesses pour rendre la main à Ember. Il est normalement très bien équipé pour justement faire un rendu progressif de la page au fur et à mesure que les données arrivent, et là on forcerait juste le _loader_ à s'afficher plus longtemps ce qui donne à l'utilisateur une moins bonne impression de performances, par rapport à l'alternative (pas de `await`/`Promise.all`…) où la page s'affichera avec les données actuelles mais la requête sera faite en tâche de fond et _si_ elle remonte des données différentes de ce qu'il y a en cache, alors la page se mettra à jour.

Peut-être dans ce cas-ci ne veut-on absolument pas prendre le risque d'afficher provisoirement des données pas à jour, peut-être pas. Je voulais juste rappeler qu'il existe des cas valides pour exécuter un `reload` sans attendre sa résolution.

D'ailleurs en ce qui concerne le "on a vu que ne pas avoir de `return` pouvait perturber ember", ça serait peut-être pas mal d'avoir une règle un peu plus étayée non ? Sinon j'ai peur qu'on se mette à programmer par supersition 🤔 . Il y a toujours un `return undefined` implicite à la fin des fonction JavaScript donc c'est compliqué de dire qu'il n'y a _pas de return_. On peut éventuellement distinguer le fait de retourner une promesse plutôt que `undefined`, mais ici comme la fonction est marquée `async` il y a un `return Promise.resolve(undefined)` implicite à la fin : la fonction ne peut pas faire autre chose que retourner une promesse. Je me demande donc ce qui pourrait "perturber" Ember ?

Reply:
Si l'on souhaite que Ember affiche un loader pour la route pendant le chargement de ces données, il est nécessaire de lui renvoyer une promesse. Ce qui lui permettra d'afficher / cacher le loader une fois la promesse résolue. Je pense que c'est en ce sens qu'il serait perturbé s'il n'a pas de promesse.

Reply:
Ce que je disais, en plus court, c'est :
 * on n'a pas toujours envie d'attendre le chargement des données pour afficher la page, et dans ce cas c'est OK de ne pas attendre les `reload` et de renvoyer une promesse déjà résolue ;
 * une fonction marquée `async` _ne peut pas_ renvoyer autre chose qu'une promesse, même sans `return` explicite.